src/atomate2/qchem/files.py

```diff
@@ -0,0 +1,153 @@
+        will be inferred from the current user. If ``None``, the local filesystem will
+        be used as the source.
+    additional_qchem_files : list of str
+        Additional files to copy.
+    file_client : .FileClient
+        A file client to use for performing file operations.
+    """
+    src_dir = strip_hostname(src_dir)  # TODO: Handle hostnames properly.
+    
+    logger.info(f"Copying QChem inputs from {src_dir}")
+    opt_ext = get_largest_opt_extension(src_dir, src_host, file_client=file_client)
+    directory_listing = file_client.listdir(src_dir, host=src_host)
+
+    # find required files
+    files = ("mol.qin", "mol.qout") + tuple(additional_qchem_files)
+    required_files = [get_zfile(directory_listing, r + opt_ext) for r in files]
+
+    copy_files(
+        src_dir,
+        src_host=src_host,
+        include_files=required_files,
+        file_client=file_client,
+    )
+
+    gunzip_files(
+        include_files=required_files,
+        allow_missing=True,
+        file_client=file_client,
+    )
+
+    # rename files to remove opt extension
+    if opt_ext:
+        all_files = required_files
+        files_to_rename = {
+            k.name.replace(".gz", ""): k.name.replace(opt_ext, "").replace(".gz", "")
+            for k in all_files
+        }
+        rename_files(files_to_rename, allow_missing=True, file_client=file_client)
+
+    logger.info("Finished copying inputs")
+
+@auto_fileclient
+def get_largest_opt_extension(
+    directory: Path | str,
+    host: str | None = None,
+    file_client: FileClient | None = None,
+) -> str:
+    """
+    Get the largest numbered opt extension of files in a directory.
+
+    For example, if listdir gives ["qchemrun.xml.opt1.gz", "qchemrun.xml.opt2.gz"],
+    this function will return ".opt2".
+
+    Parameters
+    ----------
+    directory : str or Path
+        A directory to search.
+    host : str or None
+        The hostname used to specify a remote filesystem. Can be given as either
+        "username@remote_host" or just "remote_host" in which case the username will be
+        inferred from the current user. If ``None``, the local filesystem will be used.
+    file_client : .FileClient
+        A file client to use for performing file operations.
+
+    Returns
+    -------
+    str
+        The opt extension or an empty string if there were not multiple relaxations.
+    """
+    opt_files = file_client.glob(Path(directory) / "*.opt*", host=host)
+    if len(opt_files) == 0:
+        return ""
+
+    numbers = [re.search(r".opt(\d+)", file.name).group(1) for file in opt_files]
+    max_relax = max(numbers, key=lambda x: int(x))
+    return f".opt{max_relax}"
+
+
+def write_qchem_input_set(
+    molecule: Molecule,
+    input_set_generator: QChemInputGenerator,
+    directory: Union[str, Path] = ".",
+    from_prev: bool = False,
+    **kwargs,
+):
+    """
+    Write QChem input set.
+
```

Comment:
I don't understand why this method is needed. Shouldn't this always be handled by `InputSet.write_input`? If there is a need to instantiate a calculation from a previous directory, shouldn't that also be handled within the corresponding generator?

Reply:
The InputSet.write_input defines the way to write the input. But at some point you have to actually call it. And this function is the place where the function is called.